Code/Framework/AzCore/AzCore/Math/Capsule.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Vector3.h>
+
+namespace AZ
+{
+    //! A class representing a capsule (a cylinder with a hemispherical cap at each end).
+    class Capsule
+    {
+    public:
+        AZ_TYPE_INFO(Capsule, "{4B7E9154-B258-40D2-81A4-A27A25030588}");
+
+        Capsule() = default;
+        Capsule(const Vector3& firstHemisphereCenter, const Vector3& secondHemisphereCenter, float radius);
+
+        //! Gets the centre of the first hemispherical cap (the order of the caps is arbitrary).
+        const Vector3& GetFirstHemisphereCenter() const;
+
+        //! Gets the center of the second hemispherical cap (the order of the caps is arbitrary).
+        const Vector3& GetSecondHemisphereCenter() const;
+
+        //! Gets the overall center of the capsule.
+        Vector3 GetCenter() const;
+
+        //! Gets the radius of the capsule.
+        float GetRadius() const;
+
+        //! Gets the height of the cylindrical section of the capsule.
+        float GetCylinderHeight() const;
+
+        //! Gets the total height of the capsule, including the cylindrical section and both hemispherical caps.
+        float GetTotalHeight() const;
+
+        //! Sets the center of the first hemispherical cap (the order of the caps is arbitrary).
+        void SetFirstHemisphereCenter(const Vector3& firstHemisphereCenter);
+
+        //! Sets the center of the second hemispherical cap (the order of the caps is arbitrary).
+        void SetSecondHemisphereCenter(const Vector3& secondHemisphereCenter);
+
+        //! Sets the radius of the capsule.
+        void SetRadius(float radius);
+
+        //! Returns whether this capsule is identical to another within the tolerance (allowing either order for the caps).
+        bool IsClose(const Capsule& rhs, float tolerance = Constants::Tolerance) const;
+
+    private:
+        Vector3 m_firstHemisphereCenter; //!< The center of one of the hemispherical ends (order is interchangeable). 
+        Vector3 m_secondHemisphereCenter; //!< The center of the other hemispherical end (order is interchangeable).
+        float m_radius; //!< The radius of the cylindrical and hemispherical sections of the capsule.
```

Comment:
These variables should be initialized.
Otherwise if some does
```
AZ::Capsule capsule;
capsule.GetRadius();
```

That will return uninitialized data.

Reply:
The whole math library is like that - I think you can do AZ::Vector3 vector3; vector3.GetX(); with similar results. I'm not a fan of that myself, but I think it was meant as an optimization.

Reply:
Then how about we make the default constructor to prevent the uninitialized case?
And if the user wants an uninitialized capsule we can provide a static function that provides one
i.e

```
class Capsule
{
public:
    static Capsule CreateUninitialized();
private:
    Capsule() = default;
};
```

That way we don't pay the "performance" hit by always initializing a capsule via default constructor, but we get a measure of safety by preventing users from inadvertently creating an uninitialized capsule.

Reply:
I'm afraid I'm with @greerdv on this. This is how the rest of the whole math library works - the proposed alternative is nice and I do like that idea but we'd need to change this wholesale to be consistent, so might be something we do in a separate PR, I don't think the current API should block the PR

Reply:
Wouldn't that make it more complicated to have for example a vector of capsules? 
`AZStd::vector<Capsule>` would that work having default constructor private?

Since the whole math lib is done this way, maybe this could be a separate discussion?

Reply:
Yeah, I like your suggestion way better. If we're going to do that though, I'd like to do it to the whole math library for consistency (in a separate change).

Reply:
Actually I don't think we should let "perfect" be the enemy of "good".

Just because the rest of the Math Library has an API that could improved, doesn't mean we have to make the new Capsule class use an an inferior API just for consistency.

Reply:
OK, I've made the update to the Capsule class.

Reply:
No problem, and that uninitialized thing has always bugged me, it's just that it was an established pattern which was hard to fight against, but you are right.